os/fs/driver/mtd/smart.c

```diff
@@ -503,6 +503,28 @@ static const struct block_operations g_bops = {
  * Public Functions
  ****************************************************************************/
 
+#ifdef CONFIG_DEBUG_FS
+static int smart_dumpsector(FAR struct smart_struct_s *dev, uint16_t physsector)
+{
+	int ret = OK;
+	ret = MTD_BREAD(dev->mtd, physsector * dev->mtdBlksPerSector, dev->mtdBlksPerSector, (FAR uint8_t *)dev->rwbuffer);
+	if (ret != dev->mtdBlksPerSector) {
+		fdbg("Error reading phys sector %d\n", physsector);
+		return -EIO;
+	}
+
+	fsdbg("Sector dump physical sector : %d sector size : %d\n", physsector, dev->sectorsize);
+	for (int i = 0; i < dev->sectorsize; i++) {
+		fsdbg("[%02x]", (FAR uint8_t)dev->rwbuffer[i]);
```

Comment:
Write entire sector's contents to dev->rwbuffer?
This will be followed by data being picked up from dev->rwbuffer?

Reply:
hmmm... because I added it for debug when you faced problem as a dump after assert, it never working like a internal process of smartfs, If we faced any problem then I will change it, I think its not necessary for now